plugin/src/main/scala/org/virtuslab/akkasaferserializer/SerializerTrait.scala

```diff
@@ -0,0 +1,3 @@
+package org.virtuslab.akkasaferserializer
+
+class SerializerTrait extends scala.annotation.ConstantAnnotation
```

Comment:
I'm testing that now... there's a small problem here, this trait should be both on the classpath of the plugin AND of the project itself ;) so just adding `compilerPlugin("org.virtuslab" %% "safer-serializer-plugin" % "0.1.0-SNAPSHOT")` is not enough, I ended up with `scala.ScalaReflectionException: object org.virtuslab.akkasaferserializer.SerializerTrait in compiler mirror not found.`

Reply:
Probably `SerializerTrait` should then be extracted to a separate project... or maybe, the FQCN of the trait can be passed via `-P...` option to Scala compiler instead?

Reply:
I think annotation with double import is the cleanest option. I believe annotation is easier to use than `-P` scalac option and separating plugin form annotation neither reduce boilerplate code nor improve functionality.

Reply:
Okay! lemme try now then...

Reply:
Under Scala 2.13, I'm getting a false positive error:

```
[error] /home/plipski/hydra-backend/services/src/main/scala/hydra/wallets/WalletActorProtocol.scala:21:18: hydra.wallets.WalletActorProtocol.commands.WalletCommand does not extend annotated trait
[error]     sealed trait WalletCommand extends CborSerializable {
[error]                  ^
```

Reply:
Where `CborSerializable` is defined as:

```
package hydra

import org.virtuslab.akkasaferserializer.SerializerTrait

@SerializerTrait
trait CborSerializable
```

Reply:
And WalletCommand is:

```scala
private object WalletActorProtocol {

  object commands {

    sealed trait WalletCommand extends CborSerializable {
      def walletId: WalletId
      val replyTo: ActorRef[WalletResponse]
    }
// ...
```

Reply:
Not a matter of `private object`, with just `object` it's still the same false-positive error... maybe a matter of nested objects?

Reply:
I think the plugin handles partial compilation very badly. Try `sbt clean` before `sbt compile`

Reply:
Hmmm no change after a clean... maybe you could add some additional log messages to the plugin? So that I can provide more details about what's important in the context

Reply:
Okay, seems resolved now, not sure which change fixed that :)